BMS tesseracted

mjmagee991 opened the pull request:
- Added BQ reset pin
- Added error LED pin
- Added reset and error logic to take advantage of new pins

mjmagee991 committed:
Updated pins to v4 hardware

mjmagee991 committed:
Added reset pin

mjmagee991 committed:
Added error LED

mjmagee991 committed:
New GitHub actions file

aclowmclaughlin reviewed (approved):
Looks good- is there a reason that the bq reset wasn't implemented initially?

Not important just curious.

mjmagee991 committed:
Fixed thermistor interpretation in BQ interface

ActuallyTaylor commented:
This has been validated on BMS v4 1 & 2. It should be good for merging now.

ActuallyTaylor reviewed (approved):
Looks good! Also runs well, I had no problems on the code end when validating the boards. Docs will come in a future PR.

ActuallyTaylor committed:
Planning the testing setup documentation.

Still needs a lot more work, but it turns out writing good docs is difficult. I will continue to work on this throughout the semester.

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor committed:
Update workflow to find specific problems

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor committed:
Voltage documents

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor committed:
Finished polarity docs + added a new polarity cmd.

I finished up the documentation for checking the polarity. There is a paragraph style explanation with more detail and a bulleted list that is easier to follow. I aslo added a new command into the bq_interface that will read the current and convert it for you. This removes the need to use a direct memory read and then convert the hexadecimal.

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor committed:
Only build CI for the f334 as thats the only board we care about for v4.

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor committed:
added archive so i can inspect elfs

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor committed:
always run archives cause it always failes

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor committed:
remove -whnodev

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor committed:
Updated EVT-Core and Namespaces

Updated EVT-Core to the latest version, then renamed the DEV namespace to dev because it fits the new EVT core standard

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor commented:
I am working on analyzing were the increased FLASH usage is coming from. At the moment, it appears to be a problem with the CI using the newest arm tools, and our development using version 12. 

I have also noticed that after updating EVT-Core it seems that even arm tools 12 creates binaries just too big.

It also seems like building for release, consistently adds ~1k extra flash over the debug releases...


Here are elf analysis of 3 build configurations (Debug, Release & CMAKE Build All). These are HTML files generated using an [elf analysis tool](https://github.com/jedrzejboczar/elf-size-analyze):  [BMS Elf Analysis.zip](https://github.com/user-attachments/files/25118751/BMS.Elf.Analysis.zip)

ActuallyTaylor committed:
Better polarity photo

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor committed:
Change optomization flags.

Originally, we were always using -Og which optomizas for debugging which increases binary size.

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor committed:
CI/CD Release

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor committed:
Merge branch 'feature/mjmagee991/v4-bringup' into feature/actuallytaylor/v4-validation-docs

ActuallyTaylor committed:
Remove other builds

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor committed:
Use Os for debug?

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor committed:
strip flags

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor committed:
FLASHHHHHH

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor committed:
Updated build flags for size

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor committed:
Use const reference to reduce copy

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor committed:
Change the flags

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor committed:
Reduce can queue size

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor committed:
put back queue size

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor committed:
Fixed CANopen

This fixes CANopen for the bms. The initialization of CANopen was happening before the board was connected to the CAN network. This led it to just never setting up.

Signed-off-by: Taylor Lineman <git@actuallytaylor.com>

ActuallyTaylor commented:
The DEV1-BMS target will not build on the CI. The CI is putting out 4% bigger bin files than any physical hardware we have. We are just going to ignore this for now and fix the CI later...

ActuallyTaylor committed:
Merge branch 'feature/mjmagee991/v4-bringup' into feature/actuallytaylor/v4-validation-docs

ActuallyTaylor committed:
Merge pull request #24 from RIT-EVT/feature/actuallytaylor/v4-validation-docs

mjmagee991 committed:
Implementation cleanup

mjmagee991 committed:
TEMP COMMIT FOR INTERLOCK
